src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CreateSDORespondent1NotificationHandler.java

```diff
@@ -8,32 +8,33 @@
 import uk.gov.hmcts.reform.civil.callback.CallbackHandler;
 import uk.gov.hmcts.reform.civil.callback.CallbackParams;
 import uk.gov.hmcts.reform.civil.callback.CaseEvent;
-import uk.gov.hmcts.reform.civil.notify.NotificationsProperties;
-import uk.gov.hmcts.reform.civil.enums.CaseCategory;
 import uk.gov.hmcts.reform.civil.model.CaseData;
-import uk.gov.hmcts.reform.civil.notify.NotificationService;
-import uk.gov.hmcts.reform.civil.service.OrganisationService;
-import uk.gov.hmcts.reform.civil.prd.model.Organisation;
 
 import java.util.List;
 import java.util.Map;
-import java.util.Optional;
 
 import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
 import static uk.gov.hmcts.reform.civil.callback.CaseEvent.NOTIFY_RESPONDENT_SOLICITOR1_SDO_TRIGGERED;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.NOTIFY_RESPONDENT_SOLICITOR2_SDO_TRIGGERED;
 
+/**
+ * Can handle both respondent 1 and respondent 2 notification for sdo created.
+ */
 @Service
 @RequiredArgsConstructor
-public class CreateSDORespondent1NotificationHandler extends CallbackHandler implements NotificationData {
+public class CreateSDORespondent1NotificationHandler extends CallbackHandler {
```

Comment:
To avoid having to handle both respondent 1 and 2 in here with various if statements, it would be better to make this an abstract class and have two new handlers - one for respondent1 and the other for respondent 2 (perhaps further subclasses for lip and lr if possible?) You can then set the TASK_ID and Notification Sender fields as appropriate in the sub classes without the need for if statements.

Reply:
The subclasses for lip and lr are precisely the *Sender classes. I think that creating two other classes to avoid two ifs is a bit overkill, but all right, I'll split this class again.

Reply:
I think what Paul meant was the handlers. you combined two handlers into one class. that is now confusing because that handler states respondent1 but addresses both. What Paul meant i think is have an abstract class and have two handlers for respondent1 and respondent2 inherit that class. so that responsibility is split and we dont have so many if statements

Reply:
Yeah, having to use if statements to switch behaviour between two very similar objects is a code smell and indicative of non-OO design. With subclasses and the single responsibility pattern, you are encapsulating the logic in the right places - the calling classes shouldn't need to know this logic or what needs to be called for each variation, much better to use polymorphism properly. An added benefit is it will be easier to add/change subclasses in future, for example if we change to use arrays for multi party respondents etc.